tests/components/iotty/test_cover.py

```diff
@@ -0,0 +1,233 @@
+"""Unit tests the Hass COVER component."""
+
+from aiohttp import ClientSession
+from freezegun.api import FrozenDateTimeFactory
+from iottycloud.verbs import (
+    OPEN_PERCENTAGE,
+    RESULT,
+    STATUS,
+    STATUS_CLOSING,
+    STATUS_OPENING,
+    STATUS_STATIONATRY,
+)
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    DOMAIN as COVER_DOMAIN,
+    SERVICE_CLOSE_COVER,
+    SERVICE_OPEN_COVER,
+    SERVICE_SET_COVER_POSITION,
+    SERVICE_STOP_COVER,
+    STATE_CLOSED,
+    STATE_CLOSING,
+    STATE_OPEN,
+    STATE_OPENING,
+)
+from homeassistant.components.iotty.const import DOMAIN
+from homeassistant.components.iotty.coordinator import UPDATE_INTERVAL
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from .conftest import test_sh_one_added
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+
+async def test_open_ok(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    local_oauth_impl: ClientSession,
+    mock_get_devices_twoshutters,
+    mock_get_status_filled_stationary_0,
+    mock_command_fn,
+) -> None:
+    """Issue an open command."""
+
+    entity_id = "cover.test_shutter_0_test_serial_sh_0"
+
+    mock_config_entry.add_to_hass(hass)
+
+    config_entry_oauth2_flow.async_register_implementation(
+        hass, DOMAIN, local_oauth_impl
+    )
```

Comment:
Not sure why this is here

Reply:
Since this is a test which basically only mocks the APIs, the iotty domain needs to be registered for hass to work correctly with the integration

Reply:
Can you explain more, what happens without it? I've never had this with any oauth based integration, so I'm not sure why it's needed here

Reply:
Without it I get this "Implementation not available" error

```ERROR    MainThread homeassistant.config_entries:config_entries.py:671 Error setting up entry IOTTY00001 for iotty
Traceback (most recent call last):
  File "/workspaces/home-assistant-core/homeassistant/config_entries.py", line 587, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/home-assistant-core/homeassistant/components/iotty/__init__.py", line 39, in async_setup_entry
    implementation = await async_get_config_entry_implementation(hass, entry)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/home-assistant-core/homeassistant/helpers/config_entry_oauth2_flow.py", line 431, in async_get_config_entry_implementation
    raise ValueError("Implementation not available")
ValueError: Implementation not available```

Reply:
In Withings I have a fixture that auto imports credentials, do you have something similar?

Reply:
Something similar, yeah, I have a mock for `mock_config_entry` (line 100 of `conftest.py`) which has the auth information filled